spark/spark-tensorflow-distributor/tests/integration/test_mirrored_strategy_runner.py

```diff
@@ -100,3 +101,21 @@ def train_fn():
     assert runner.get_num_tasks() == 2
     gpus_used_by_each_task = runner.run(train_fn)
     assert gpus_used_by_each_task == [0, 0]
+
+@pytest.mark.parametrize('num_workers', [2], indirect=True)
+@pytest.mark.parametrize('num_gpus_per_worker', [4], indirect=True)
+def test_get_gpus_owned_in_spark_task(num_workers, num_gpus_per_worker):
```

Comment:
You can set CUDA_VISIBLE_DEVICES in the test env, and then test the user-facing behavior here instead of a private method.